### DISCUSSION TITLE
Using load_data_type_definitions with multiple servers

### DISCUSSION THREAD

**Discussion by goudcode** [FreeOpcUa/opcua-asyncio#1277] [Q&A] [answered]
We are trying to connect to multiple PLC's to combine some data in a python script.
We managed to get this working by calling the load_data_type_definitions method each time we talk to another PLC.

The problem we are facing is that this will cost too much time.
Is it possible to load the data types for each individual client instance instead of loading the types in a global space?

Thanks.

**Comment by schroeder-** [comment]
If they have the same nodeids and namespaces, for there datatypes, you only need to call load_data_type_definitions, once.

**Reply by goudcode** [reply]
They have the same nodeids and namespaces, but are different types.
The PLC's are modules with the same interface, but communicate different types.
That is why we would like to read the data types for each client.

When we call the load_data_type_definitions on every client, we have found out that the types are loaded in a global scope.
Is it possible to load the types into the scope of a specific client?

**Reply by schroeder-** [reply, marked as answer]
Currently this is not possible. Either you use the multiprocess module to isolate your clients or you propose a pr for a client depended solution.

**Reply by goudcode** [reply]
Thanks for taking the time to clarify this.
We are trying the multiprocessor way, if we come up with a more elegance way to have the types per client we will propose a pr.
